zerver/views/auth.py

```diff
@@ -658,13 +661,22 @@ def redirect_to_deactivation_notice() -> HttpResponse:
 
 
 def update_login_page_context(request: HttpRequest, context: Dict[str, Any]) -> None:
-    for key in ("email", "already_registered", "is_deactivated"):
+    for key in ("email", "already_registered"):
         try:
             context[key] = request.GET[key]
         except KeyError:
             pass
 
-    context["deactivated_account_error"] = DEACTIVATED_ACCOUNT_ERROR
+    deactivated_email = request.GET.get("is_deactivated")
+    if deactivated_email is None:
+        return
+    try:
+        validate_email(deactivated_email)
+        context["deactivated_account_error"] = mark_safe(
+            DEACTIVATED_ACCOUNT_ERROR.format(username=deactivated_email)
+        )
+    except ValidationError:
+        logging.info("Invalid email in is_deactivated param to login page: %s", deactivated_email)
 
```

Comment:
`deactivated_email` is an HTTP GET parameter under the control of someone creating a link; is it safe to render it into the `context` with `mark_safe`, or could this be used as an XSS attack vector?

Reply:
Hmm, this was secured by the `validate_email` call - any XSS attempt should fail because the necessary characters are forbidden in emails - but doesn't hurt to be more careful here so I added an `escape` call.